member-service/src/main/java/com/example/memberservice/member/model/entity/Members.java

```diff
@@ -95,24 +96,73 @@ private Members(String nickName, String code, String email, String phoneNumber,
         this.provider = provider;
         this.providerId = providerId;
 
-        this.canWork = false;
+        this.role = MemberRole.NONE;
     }
 
     // 닉네임 변경
     public void updateNickName(String newNickName) {
         this.nickName = newNickName;
     }
 
-    //판매자 등록 가능 여부 확인
-    public boolean canEnableWork() {
-        return Boolean.FALSE.equals(this.canWork);
+//    //판매자 등록 가능 여부 확인
+//    public boolean canEnableWork() {
+//        return Boolean.FALSE.equals(this.canWork);
+//    }
+//
+//    // 판매자 등록 변경
+//    public void updateCanWork(Boolean canWork) {
+//        this.canWork = canWork;
+//    }
+
+    public boolean canRegisterClientState() {
+        return (role.equals(MemberRole.NONE) || role.equals(MemberRole.FREELANCER));
```

Comment:
enum을 비교할 때 `equals`를 사용하셨군요👓 

enum의 `equals`는 내부적으로 `==`와 동일하게 동작합니다.

```java
    public final boolean equals(Object other) {
        return this==other;
    }
```
참고로, NPE를 방지하기 위해서는 상수를 먼저 작성하는 방식이 좋다고 하네요👍 저는 보통 아래와 같이 사용합니다.

```suggestion
        return MemberRole.NONE == role || MemberRole.FREELANCER == role;
```

Reply:
enum 의 경우 == 과 동일하게 동작하는 것은 알고 있었지만 
enum도 하나의 객체로 보고 최대한 메서드를 활용하는 것이 좋다 생각해
NPE가 터지지 않는 상황에선 == 보단 .equals()를 선호하여 사용하고 있었는데
equals이 아닌 == 이 표준인 것은 처음알았네요! 

항상 표준 혹은 네이밍과 관련하여 많은 점을 배워가고 있습니다!
이점을 이후 작업 간 참고하여 코드를 작성하겠습니다.

> NPE를 방지하기 위해서는 상수를 먼저 작성하는 방식이 좋다고 하네요

이와 관련하여 기본적 자바 문법 이야기이지만
혹시 == 연산의 경우 상수가 뒤에 가면 NPE가 발생하나요?

.equals 의 경우에는 변수가 NULL 이면 NPE 가 터지겠지만 == 의 경우에는 아닌 것으로 알고 있었는데 잘못 알고 있는 점일까요! 
혹 아니면 이것도 기술적보다는 표준적인 것일까요!

Reply:
`MemberRole.NONE.equals(role)`을 말씀하신 것 같습니다. 이러면 NPE가 안터지는 것으로 알고 있습니다.

NPE를 터트리는가 안터트리는가는 선택에 따라 달라지는 듯 합니다. NPE를 안 터트릴 경우, 해당 예외가 숨겨져 디버깅에 어려움을 겪을 수도 있다는 것 같고 딱히 관계 없으면 숨겨도 괜찮다고 하는 견해도 있는 것 같아요.